src/utils/get-config.js

```diff
@@ -12,9 +12,18 @@ export default function (id, mainPath) {
         mainPath = path.join(mainPath, CONSTANTS.configFileName);
 
     var mainDir      = path.dirname(mainPath);
-    var configString = readFileSync(mainPath).toString();
 
-    var config = JSON.parse(configString);
+    var config;
```

Comment:
I think we can do it in a more elegant way:
```js
const configExtension = path.extname(mainPath);

if (Object.keys(require.extensions).some(extention => extension  === configExtension)) 
    config = require(mainPath);
else {
    var configString = readFileSync(mainPath).toString();

    config = JSON.parse(configString);
}
```

This way we won't lose an error if it happens during `require`.

Reply:
Okay rad, I'll update the code and look to making a test for this 👍

Reply:
This doesn't work as `extension  === configExtension` is never met... As there's no extension on our configExtension (just `testcafe-electron-rc`). 

Hmmm, the original code just tries for _anything_ resolveable first, and on failure, fallsback to json... I'm not sure what we want to be doing with a require error there, as we're falling back. Perhaps we just want to print a warning so folk are aware?

Reply:
```js

    try {
        config = require(mainPath);
    }
    catch ( e ) {

        if ( !e.message.includes('Cannot find module') )

            throw e;

        var configString = readFileSync(mainPath).toString();

        config = JSON.parse(configString);
    }
```

Reply:
`require` won't load a JS config if its name doesn't end with `.js`. So, if you want to use JS, you *will have* to explicitly specify the `.js` extension.

Honestly, I consider this naming decision (`.testcafe-electron-rc` without the explicit `.json` extension) as my mistake. If you explicitly specify an extension, you will get at least basic autocompletion in IDEs like VS Code and WebStorm. And maybe someday we will add a JSON schema and submit it to https://schemastore.org to get the full autocompletion.

So, I still prefer the way I described. In my opinion, using explicit extensions is the right way, and if there is no extension or if it is an unknown one, we should try the current legacy way with file reading and JSON parsing.

Reply:
>if there is no extension or if it is an unknown one, we should try the current legacy way with file reading and JSON parsing.
But we can try both `js` and default to `json` when using a directory too. I don't see why _not_ to try requiring `.js` files (or other supported formats in future)...

I had toyed with the idea of looping over `require.extensions` to do just that, but actually [it seems like we should _not_ be using this, as it's been marked `deprecated` in latest node :( ](https://nodejs.org/api/modules.html#modules_require_extensions)

Reply:
I understand that it seems dangerous using a `deprecated` functionality. However, IMHO it is deprecated since it is a very low-level feature, rather than an obsolete one. There is no other way to control `require`'s behaviour for different file types, so even big projects like `ts-node` use it. Moreover, the TestCafe's compiler subsystem heavily relies on this feature. However, I won't insist if you don't want to play with it 😄

Reply:
I can re add this if you want to be using it. I just think it may be a headache for removing it later. 

I'm also of the opinion that _at least_ `.js` files should be detected and as seamless as the base `testcafe-electron-rc` (ie, we shouldn't _need_ to specify it when running tests). So if you're happy doing that, it probably makes sense to do that for _all_ extensions present in `require.extensions`, no? 

Let me know what you think and I can get the PR updated :+1: